packages/gatsby/src/utils/local-eslint-config-finder.js

```diff
@@ -0,0 +1,23 @@
+const path = require(`path`)
+const glob = require(`glob`)
+
+module.exports = directory => {
+  let hasEslintConfig = false
+
+  try {
+    const pkg = require(path.resolve(directory, `package.json`))
+    if (pkg.eslintConfig) {
+      hasEslintConfig = true
+    }
+  } catch (error) {
+    // not sure what we should do here...
+  }
+
+  const eslintFiles = glob.sync(`.eslintrc*`, { cwd: directory })
```

Comment:
this might produce some false positives if someone tries to rename their `.eslintrc` to something like for example `.eslintrc.tmp` - I think we should use more strict glob here to find exactly those files that eslint would use:

https://github.com/eslint/eslint/blob/a370da27a7957e2a5e038f75cd97c0951f8c4ddc/lib/config/config-file.js#L46-L53  :
```js
const CONFIG_FILES = [
    ".eslintrc.js",
    ".eslintrc.yaml",
    ".eslintrc.yml",
    ".eslintrc.json",
    ".eslintrc",
    "package.json"
];
```

Reply:
i didn't even think of that! good call, updating now

Reply:
I wonder if this should check up the directory structure...I often have a eslintrc in a root directory above the site...like the gatsby repo, where `www` would use `../` eslint

Reply:
Hmm maybe we could use this https://eslint.org/docs/developer-guide/nodejs-api#clienginegetconfigforfile to figure out if eslint would pickup any config and don't play guessing game here?

Reply:
Hmm yeah. This could be confusing if you have an eslint config way up the directory tree though.

With a config file at:

```
~/.eslintrc
```

and a Gatsby site at:

```
~/my-stuff/dev/client-projects/client-a/gatsby-project
```

Maybe that's ok though? An alternative could be to search up the directory tree until you find a `.git` directory and then stop? But maybe that's trying to be too clever.

Reply:
> search up the directory tree until you find a .git directory and then stop?

that is not a bad default i don't think 👍 compared to just checking the gatsby dir, ideally we'd be able to use whatever eslint uses, but the issue there is that is very much dependent on the `cwd` when calling eslint. Maybe we should use whatever logic editor extensions use? that may just be "check the project root" specfic to like vscode, but maybe it's more clever?